src/internal/endpoint.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * Represents an AWS service endpoint, providing utilities for parsing and handling URL details.
+ */
+export class Endpoint {
+    private _protocol: string
+    private _hostname: string
+    private _port?: number
+
+    // Default protocol, this can be globally changed as per application requirements
+    private static readonly DEFAULT_PROTOCOL = 'https' // Set this as per AWS.config
+
+    /**
+    //  * Constructs a new Endpoint instance.
+     *
+     * @param {string} endpoint - The URL to construct an endpoint from. If the URL omits a protocol, the default protocol will be used.
+     */
+    constructor(endpoint: string) {
+        const isDefaultProtocol =
```

Comment:
I think this constructor can be simplified a bit if you leverage `URL`.  

```ts
const completeUrl = endpoint.startsWith('http://') || endpoint.startsWith('https://')
    ? endpoint
    : `${Endpoint.DEFAULT_PROTOCOL}://${endpoint}`;

const url = new URL(completeUrl);

this._protocol = url.protocol.slice(0, -1);
this._hostname = url.hostname;
this._port = url.port ? parseInt(url.port) : undefined;
```

Reply:
Unfortunately `URL` is not yet available in the k6 JS runtime, hence doing it manually instead 👍🏻

Reply:
👍 . Heads up, you might want to remove
https://github.com/grafana/k6-jslib-aws/pull/61/files/6d50785f5a9123ff5c2ffca766fa8b5fe4969256#diff-cb88677cfb485379a627afd78775c5daebe4bf9085f172f84a87549d8b62e31fR97

Reply:
Good catch 👀